### ISSUE TITLE
Testing for customer.deleted generates a PHP notice

### ISSUE THREAD

**Issue by acabal** [stripe/stripe-php#73]
If a customer object is deleted, subsequent responses from Stripe will include a customer.deleted property.  This property only exists if the customer is deleted--if a customer is NOT deleted, the property will not be "false", but rather unset.

When testing for this property in an if statement for a customer who is NOT deleted, PHP generates a warning:

```
$customer = Stripe_Customer::retrieve('cus_abc123');  //this customer has NOT been deleted
if($customer->deleted){
    /* do something */
}

Stripe Notice: Undefined property of Stripe_Customer instance: deleted
```

Which is technically correct, since the property is not set in the JSON response.  However this clogs our logs files.  Since there's no way to test if the property is unset (for example, calling isset($customer->deleted) generates the same warning), and no obvious way to access the raw JSON response from an object retrieved through the API, there's no obvious workaround for testing for customer.deleted without generating PHP warnings.

**Comment by mr-mmmmore** [comment]
I have encountered the same problem and don't understand why the issue was closed with no solution added to the Stripe lib. We can use property_exists or $customer->offsetExists, but these are only fixes to a unexpected behavior.

I find there is something wrong about this `deleted` property: it looks as if it was an undocumented property. It is not listed in the Customer object in the API reference, it causes error when referenced and not set...

But it is still a documented property, it appears in the doc, so it should be fully supported:

- it should be listed in the doc in the customer object properties,
- it should have a default value of `false` in the object and shouldn't cause notices.

**Comment by ob-stripe** [comment]
Hi @mr-mmmmore, and thanks for the feedback.

What you describe is actually expected (if unintuitive) behavior: Stripe's API only returns `"deleted": true` for deleted resources, but it does not return a `"deleted"` attribute at all for non-deleted resources. At this time, Stripe's PHP library dynamically populates attributes from API responses, so if the API response does not include a `"deleted"` attribute then the PHP object will in turn not have a `deleted` property.

That said, I agree that using `property_exists()` is ugly and undesirable. I've opened a PR that adds a `isDeleted()` method that is always available: https://github.com/stripe/stripe-php/pull/502.

**Comment by ob-stripe** [comment]
The PR was merged and released in version 6.13.0. You can now call `isDeleted()` on any Stripe object instance, and it will return the value of the `deleted` attribute if the object has this attribute, or `false` otherwise.

**Comment by mr-mmmmore** [comment]
OK it's a lot better than using `property_exists`, thanks @ob-stripe!

**Comment by MHCreations** [comment]
> The PR was merged and released in version 6.13.0. You can now call `isDeleted()` on any Stripe object instance, and it will return the value of the `deleted` attribute if the object has this attribute, or `false` otherwise.

@ob-stripe  dumb question but would this mean that I can check a customer is deleted with 

```
$check = $stripe->customers->retrieve($customerId);
if($check->isDeleted()){
    // yes customer is deleted
}
```

??

**Comment by ggedde** [comment]
After updating some composer packages it seems that isDeleted() no longer exists. does anyone know if this has changed. If so, what is the new method to know if the Customer you get back has been deleted or active, etc?
Thanks
